projects/driver_controls/src/throttle.c

```diff
@@ -0,0 +1,159 @@
+// The module operates based on the voltage readings that come from the pedals.
+// The pedal occupies two channels of the ADS1015 one of which is chosen to be the "main" channel.
+// Everytime ADS1015 finishes a conversion of the pedal input on the main channel,
+// prv_flag_update_callback is called to set reading_updated_flag accordingly which determines if
+// the readings are up to date. A periodic safety check, prv_raise_event_timer_callback, checks
+// for this flag and if ok, it updates the position of the pedal in storage and
+// raises the events related to the pedal's position (braking, coasting, and accelerating zones).
+// The reading_ok_flag basically holds the actual state of reading_updated_flag when it is reset.
+// It also verifies if the second channel is in sync with the main channel to ensure the validity
+// of the data.
+// If the data turns out to be stale or channels aren't synched, a timeout event will be raised.
+// The throttle_get_position function simply reads the position from storage.
+#include "throttle.h"
+#include <string.h>
+#include "ads1015_def.h"
```

Comment:
Why do you need access to `ads1015_def.h`?

Reply:
I used a macro from it for initializing reading variables.

Reply:
You really shouldn't. `ads1015_def.h` is an internal header for the `ads1015` module. It should not be accessed outside of its module.